From c9e3dbad3c861e95c3c9f8818441ccb0ad98afa8 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Thu, 5 Dec 2024 15:59:12 +0100
Subject: [PATCH 1402/1695] drivers: stm32_ipcc: fix st,proc-id offset value

As specified in st,stm32-ipcc.yaml bindings: st,proc-id could take the
value 0 or 1. Fix this condition to respect the bindings.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 2dab886f3f42354d11fae27dfffa80a66468c8f0 drivers: stm32_ipcc: fix st,proc-id offset value]
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I371cfedb0591f10f4d154987ffd5f63068edfbbc
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/423178
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Michel JAOUEN <michel.jaouen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/stm32_ipcc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/drivers/stm32_ipcc.c b/core/drivers/stm32_ipcc.c
index a61369e1b..43c2e853f 100644
--- a/core/drivers/stm32_ipcc.c
+++ b/core/drivers/stm32_ipcc.c
@@ -674,7 +674,7 @@ static TEE_Result parse_dt(const void *fdt, int node, struct ipcc_pdata *ipcc_d)
 	 * according to proc ID mailbox lbase. rbase mailbox are swapped.
 	 */
 	(void)fdt_read_uint32(fdt, node, "st,proc-id", &proc_id);
-	if (proc_id == 2) {
+	if (proc_id == 1) {
 		ipcc_d->lbase = ipcc_d->base + IPCC_C2CR;
 		ipcc_d->rbase = ipcc_d->base + IPCC_C1CR;
 	} else {
-- 
2.47.2

